ADR/0004-component-image-location.md

```diff
@@ -0,0 +1,107 @@
+* Security:
+  * Leverage short-lived Quay.io API tokens for repo/org management on Quay.io
+  * Leverage robot accounts for pushing images from CI/Build pipelines.
+
+
+
+## Decision
+
+### What
+
+* Per workspace, setup a new org “quay.io/unique-org-david/…”
+* Per component, setup a new new repo “quay.io/unique-org-david/unique-component-repo”
+* Use User’s Quay.io API token to manage the org/repo. Short-term, we'll use a pre-configured Quay.io API token associated with AppStudio to create the org/repo till we 
+ figure out how to determinstically map a user in AppStudio to a user in Quay.io.
+* Generate a robot account token scoped to the relevant repository and persist it in the user's workspace for the image build and push process to consume.
+
+
+### How
+
+#### Long-term
+* (One-time) Use an OAuth flow to get David’s Quay.io API token if not previously pulled
+* Create a new Quay.io org owned by David using David’s API token “quay.io/unique-org-david”
+* Create a new image repo for “foo” using David’s API token “quay.io/unique-org-david/appname-foo”
+* Optionally, if needed, make the image repo public using the David’s API token for easy image pulls.
+* Configure the robot account “appstudio” in “quay.io/unique-org-david” to be able to push to “quay.io/unique-org-david/appname-foo” Invalidate/Delete/Prune David’s API token. 
+
+
+#### Short-term
+
+##### Setup
+* Configure the a pre-configured Quay.io API token for repo/org management, also known as the OAuth Access token. From a security perspective, this would be treated as a typical service account token
+  belonging to AppStudio. This token would be shortlived and rotated *as frequently as practically possible*  .
+
+##### When a user creates a Component
+1. Create a new Quay.io org owned by David using the pre-configured AppStudio-wide API token `quay.io/unique-org-david`
+2. Create a new image repo for Component “foo” using the the pre-configured David’s API token `quay.io/unique-org-david/unique-component-foo`
+3. Optionally, if needed, make the image repo public using the API token for easy image pulls.
+4. Configure a robot account  `quay.io/unique-org-david` to be able to push to `quay.io/unique-org-david/unique-component-foo`
+
+
+##### Long-term
+
+The fundamental design does not change in the long-term except that we wouldn't bank on the pre-configured Quay.io API token for repo/org management.
+We would use the user's API token for working with the relevant Quay.io organizations and repositories.
+
+
+## Open Questions
+
+- Since the long-term goal is to have the Quay.io organizations owned by the user, how do we build a frictionless experience to map the user's 
+  account with the user's Quay.io account ?
+- Considering the above is figured out, we would need to add existing users as owners to the relevant organizations. This is a backend job that 
+  would need to be designed and executed at an appropriate time.
+
```

Comment:
```suggestion
- In the HACBS context, giving ownership of the image repo to users is not a problem for provenance, because integrity of images is based on attestations which are only treated as valid if signed by pipeline-service (by tekton chains). While users can in theory push images built on their laptop, the [release-service](../book/release-service.md) won't treat them as if they were built by the [pipeline-service](https://github.com/openshift-pipelines/pipeline-service).
```